src/meltano/core/tracking/environment.py

```diff
@@ -0,0 +1,139 @@
+    """
+    return getattr(obj, name, None) is not None
+
+
+class EnvironmentContext(SelfDescribingJson):
+    """Environment context for the Snowplow tracker."""
+
+    def __init__(self):
+        """Initialize the environment context."""
+        ci_markers = ("GITHUB_ACTIONS", "CI")
+        super().__init__(
+            "iglu:com.meltano/environment_context/jsonschema/1-0-0",
+            {
+                "context_uuid": uuid.uuid4(),
+                "meltano_version": meltano.__version__,
+                "is_dev_build": not release_marker_path.exists(),
+                "is_ci_environment": any(
+                    os.environ.get(marker, "").lower()[:1] in "1t"
+                    for marker in ci_markers
+                ),
+                "python_version": platform.python_version(),
+                "python_implementation": platform.python_implementation(),
+                **self.system_info,
+                **self.process_info,
+            },
+        )
+
+    @cached_property
+    def system_info(self) -> dict[str, Any]:
+        """Get system information.
+
+        Returns:
+            A dictionary containing system information.
+        """
+        freedesktop_data = (
+            platform.freedesktop_os_release()
+            if _safe_hasattr(platform, "freedesktop_os_release")
+            else defaultdict(type(None))
+        )
+
+        return {
+            "system_name": platform.system() or None,
+            "system_release": platform.release() or None,
+            "system_version": platform.version() or None,
+            "machine": platform.machine() or None,
+            "windows_edition": platform.win32_edition()
+            if _safe_hasattr(platform, "win32_edition")
+            else None,
+            "freedesktop_id": freedesktop_data["ID"],
+            "freedesktop_id_like": freedesktop_data.get("ID_LIKE", None),
+            "freedesktop_version_id": freedesktop_data.get("VERSION_ID", None),
+        }
+
+    @staticmethod
+    def get_process_timestamp(process: psutil.Process) -> str:
+        """Obtain the creation time of a process as a ISO 8601 timestamp.
+
+        Args:
+            process: The process to obtain the creation time from.
+
+        Returns:
+            A ISO 8601 timestamp formatted string.
+        """
+        return f"{datetime.utcfromtimestamp(process.create_time()).isoformat()}Z"
+
+    @cached_property
+    def process_info(self) -> dict[str, Any]:
+        """Obtain the process information for the current process.
+
+        Returns:
+            A dictionary containing the process information. Such as the hashed process name, pid, core counts, etc
+        """
+        process = psutil.Process()
+        with process.oneshot():
+            return {
+                "num_cpu_cores": psutil.cpu_count(),
+                "num_cpu_cores_available": self.num_available_cores,
+                "process_executable_path": process.exe() or None,
```

Comment:
@pandemicsyn called this out to me - sorry I missed it earlier.

`process_executable_path` can _definitely_ contain PII and is not permittable to be included.

Even this is PII:

`/usr/ajsteers/source/meltano`

And in addition to PII, there could be any amount of sensitive/non-public info in the path:

`/usr/ajsteers/projects/mysecretworldtakover/meltano/clone-army-analytics/`